### DISCUSSION TITLE
TP3 Punto 2.3

### DISCUSSION THREAD

**Discussion by Elias-Turner** [INGCOM-UNRN-P1/cursada-2023#114] [Preguntas y respuestas] [answered]
Chicos me pueden decir en que me estoy equivocando:

<details>
  <summary>Spoiler</summary>
  
```C
void insertar_cadena(char cadena[], char cadena2[], int capacidad2, int capacidad, char cadenaFusion[], int posicion) {
    int longitud1 = largo_seguro(cadena, capacidad);
    int longitud2 = largo_seguro2(cadena2, capacidad2);
    
    for(int i=0; longitud1 != '\0'; i++){
        cadenaFusion[i] = cadena[i];
        if (cadenaFusion[i] = posicion){
            for(int j=0; cadena2[j] != '\0'; j++)
            cadenaFusion[i] = cadena2[j];
        }
        cadenaFusion[i] = cadena[i];
    }
    cadenaFusion[longitud2 + longitud1] = '\0';
}
```
</details>

el anterior codigo reemplazaba los lugares pero el profe dice que no cumple y no encuetro la forma de hacer lo que me pide. 
Alguien me puede decir en que estoy mal

**Comment by martinvilu** [comment]
Discusiones de código en el repositorio. porfis

**Comment by martinvilu** [comment]
¿Qué estaba antes? ¿Qué fue lo corregido? ¿Qué cambió?

**Reply by Elias-Turner** [reply]
antes estaba asi:
<details>
  <summary>Spoiler warning</summary>
  
```C
void insertar_cadena(char cadena[], char cadena2[], int capacidad2, int capacidad, char cadenaFusion[], int posicion) {
    for (int i = 0; cadena[i] != '\0'; i++) {
        cadenaFusion[i] = cadena[i];
    }
    for (int j = 0; cadena2[j] != '\0'; j++) {
        cadenaFusion[posicion + j] = cadena2[j];
    }
    cadenaFusion[posicion + capacidad2 + capacidad] = '\0';
}
```
</details>
en este codigo si yo ingresaba elias y despues pedia que ingrese hola en la posicion 2 el resultado seria elhola

**Reply by martinvilu** [reply]
Elias, envia el enlace al repositorio. Así no se que te corregí tambien.

**Reply by Elias-Turner** [reply]
nono me corrigio el profe ivan en la clase practica de hoy

**Reply by martinvilu** [reply]
Le hacemos una invocación entonces @inomdedeu

**Reply by inomdedeu** [reply, marked as answer]
Dado el ejemplo qeu mencionas, la salida debia ser: elholaias.
Osea, si tenias la frase elias, insertas en la posición 2 la frase hola.

La mención que te hice, hace referencia a que te estaba faltando la parte que pisas de la primera cadanea

**Reply by inomdedeu** [reply]
Osea, estas implementando un reemplazo, no una inserción.

**Reply by Elias-Turner** [reply]
por eso esta este codigo:
void insertar_cadena(char cadena[], char cadena2[], int capacidad2, int capacidad, char cadenaFusion[], int posicion) {
    int longitud1 = largo_seguro(cadena, capacidad);
    int longitud2 = largo_seguro2(cadena2, capacidad2);
    
    for(int i=0; longitud1 != '\0'; i++){
        cadenaFusion[i] = cadena[i];
        if (cadenaFusion[i] = posicion){
            for(int j=0; cadena2[j] != '\0'; j++)
            cadenaFusion[i] = cadena2[j];
        }
        cadenaFusion[i] = cadena[i];
    }
    cadenaFusion[longitud2 + longitud1] = '\0';
}

el proble es que no encuentro el error y no se como hacerlo

**Reply by martinvilu** [reply]
Tenes que "hacer lugar" antes.

![imagen](https://github.com/INGCOM-UNRN-P1/cursada-2023/assets/56625/4e27326b-b36a-4bd9-bc03-5c942d7f50f0)

**Comment by martinvilu** [comment]
Marqué una solución, ya que la misma parece ser contestada por ella.

De forma que no quede la pregunta colgada.
